src/Core/System/Snippet/Command/ValidateTranslationFilesCommand.php

```diff
@@ -0,0 +1,261 @@
+use Shopware\Core\System\Snippet\Struct\ValidatedTranslationFileStruct;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+
+/**
+ * @internal
+ */
+#[AsCommand(
+    name: 'translation:validate-filenames',
+    description: 'Ensures translations have a country agnostic translation file as a base',
+)]
+#[Package('discovery')]
+class ValidateTranslationFilesCommand extends Command
+{
+    protected function configure(): void
+    {
+        $this->addOption(
+            'fix',
+            null,
+            InputOption::VALUE_NONE,
+            'Renames filenames to their agnostic equivalents. If more than one country-specific candidate exists for a single agnostic file, one has to be selected.'
+        );
+
+        $this->addOption(
+            'all',
+            null,
+            InputOption::VALUE_NONE,
+            'Includes the "custom" directory in the check for faulty filenames. The "extensions" option is ignored if specified.'
+        );
+
+        $this->addOption(
+            'extensions',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Restricts the search to the given extensions, if specified.',
+            '',
+        );
+
+        $this->addOption(
+            'ignore',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Excludes the specified paths relative to "src", "custom", or, if specified, the provided bundles. Values are comma-separated.',
+            '',
+        );
+
+        $this->addOption(
+            'dir',
+            null,
+            InputOption::VALUE_OPTIONAL,
+            'Searches only a specific directory for translation files.',
+            '',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $options = ValidatedTranslationFileOptions::fromInputInterface($input);
+        $fileValidator = new CountryAgnosticFileValidator();
+
+        $validatedFileStruct = $fileValidator->checkTranslationFiles($options);
+
+        if ($options->isFix() && $validatedFileStruct->getFixableFileCount() > 0) {
+            $validatedFileStruct = $this->hydrateFixingCollection($input, $output, $validatedFileStruct);
+            $fileValidator->fixFilenames($validatedFileStruct);
+        }
+
+        return $this->renderOutput($input, $output, $validatedFileStruct, $options);
+    }
+
+    private function hydrateFixingCollection(
+        InputInterface $input,
+        OutputInterface $output,
+        ValidatedTranslationFileStruct $validatedFileStruct,
+    ): ValidatedTranslationFileStruct {
+        $io = new ShopwareStyle($input, $output);
```

Comment:
`ShopwareStyle` is created in five places... please create in the `execute` method and pass it down

Reply:
Oh lol, didn't notice that. Good point 👍